HW0/P4/P4.py

```diff
@@ -30,48 +30,105 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+s_true=np.loadtxt("P4_trajectory.txt", delimiter=",",unpack=True)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+ax.plot(s_true[0,:], s_true[1,:], s_true[2,:],
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
+m=np.loadtxt("P4_measurements.txt", delimiter=",",unpack=True)
+notc=np.diag([1/rx,1/ry,1/rz])
+xbar= np.asarray(notc * np.asmatrix(m))
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+ax.plot(xbar[0,:], xbar[1,:], xbar[2,:],
+           '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+A = np.zeros(shape=(6,6))
+A[0,0]=A[1,1]=A[2,2]=1.
+A[0,3]=A[1,4]=A[2,5]=dt
+A[3,3]=A[4,4]=A[5,5]=1.-c*dt
 
+a = np.array([0.,0.,0.,0.,0.,g*dt])
+s = np.array([0.,0.,2.,15.,3.5,4.])
+
+S=np.zeros(shape=(6,K))
+S[:,0]=s
+S=np.asmatrix(S)
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
+for i in range(0,(K-1)):
+	S[:,(i+1)]=np.asarray((np.asmatrix(A)*np.asmatrix(S[:,i]))+np.asmatrix(a).T)
+S2=np.asarray(S)
```

Comment:
Cool! Perhaps setting up the variables as matrices, conducting all operations on them, and changing final matrix to an array would be cleaner code, but I'm nitpicking here.